widget_driver/test/drivable_widget_test.dart

```diff
@@ -14,57 +15,149 @@ void main() {
+    group('Driver Access: ', () {
+      testWidgets('Can access its driver', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
 
-      final testContainerDrivableWidget = TestContainerDrivableWidget(
-        environmentInfo: _mockRuntimeEnvironmentInfo,
-      );
-      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
 
-      final driver = testContainerDrivableWidget.driver;
-      expect(driver.aTestText, 'A test text');
+        final driver = testContainerDrivableWidget.driver;
+        expect(driver.aTestText, 'A test text');
+      });
     });
 
-    testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
-      when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+    group('Providable Properties: ', () {
+      testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final driver = testContainerDrivableWidget.driver;
+
+        expect(find.text('didCallTestMethod: false'), findsOneWidget);
+
+        driver.aTestMethod();
+        await tester.pump();
 
-      final testContainerDrivableWidget = TestContainerDrivableWidget(
-        environmentInfo: _mockRuntimeEnvironmentInfo,
-      );
-      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
-      final driver = testContainerDrivableWidget.driver;
+        expect(find.text('didCallTestMethod: true'), findsOneWidget);
+      });
 
-      expect(find.text('didCallTestMethod: false'), findsOneWidget);
+      testWidgets('On state change updateDriverProvidedProperties gets called', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
 
-      driver.aTestMethod();
-      await tester.pump();
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final driver = testContainerDrivableWidget.driver;
+
+        expect(driver.numberOfCallsToUpdateDriverProvidedProperties, 1);
+
+        driver.aTestMethod(); // Calls `notifyWidget` thereby causing a state update
```

Comment:
I am actually not sure if this behaviour is what we want..
If we call `notifyWidget` that just means that something internally in our driver changed.. It does not mean that thew widget changed.. so there would probably be no need for the `updateDriverProvidedProperties` to be called in this case...

We only need the `updateDriverProvidedProperties` to be called if the thing which gave us the `DriverProvidedProperties` has cahnged.. and who is this.. well that is the `widget`.. so I think that this method only needs to be called when `didUpdateWidget` is called..